### ISSUE TITLE
support for training ? 

### ISSUE THREAD

**Issue by doofin** [guillaume-be/rust-bert#316]
Hi , it seems the current codebase only support loading pretrained model,could you give me some advices on how to add this? thanks!

**Comment by guillaume-be** [comment]
Hello @doofin ,

Indeed there is no binaries to perform training with the models at this point. Note however that the architectures contain the utilities that would be required to train or finetune these models (masking, dropout,...). Examples of training scripts can be found on the upstream [tch](https://github.com/LaurentMazare/tch-rs) crate, such as [here](https://github.com/LaurentMazare/tch-rs/blob/main/examples/min-gpt/main.rs).

If you would like to implement training experiments I would suggest to experiment with them outside of the crate, or propose them as a script to be stored in `examples`. I expect you will face a few challenges since several training utilities from the Python ecosystem are not yet available for Rust, including:
- Distributed training utilities and wrappers, such as Pytorch Lightning, Fairscale, Deepspeed, or the Huggingface trainers, which may limit the size of models you will be able to train
- Language models training loop are usually complex, as can be seen [here](https://github.com/huggingface/transformers/blob/main/src/transformers/trainer.py) - re-implementing and validating this logic in Rust would require a significant amount of work
- Automatic Mixed Precision utilities
- Efficient dataloading utilities (that would implement the sampling, collating and batching strategies)
- Schedulers and hyper-parameter search

**Comment by doofin** [comment]
Thanks! It seems much more complicated than I imagine,because the training and decoding of transformers is inherently complex.
